src/dune/simple_rules.ml

```diff
@@ -57,6 +67,29 @@ let user_rule sctx ?extra_bindings ~dir ~expander (rule : Rule.t) =
     in
     let bindings = dep_bindings ~extra_bindings rule.deps in
     let expander = Expander.add_bindings expander ~bindings in
+    let action =
+      SC.Deps.interpret_named sctx ~expander rule.deps
+      |> SC.Action.run sctx (snd rule.action) ~loc:(fst rule.action) ~expander
+           ~dep_kind:Required ~targets ~targets_dir:dir
+    in
+    let action =
+      match rule.alias with
+      | None -> action
+      | Some alias ->
+        let stamp =
+          let action = Some (snd rule.action) in
+          stamp ~deps:rule.deps ~extra_bindings ~action
+          |> Digest.generic
+          |> Digest.to_string
+        in
+        let stamp_target = Path.Build.relative dir stamp in
+        let () =
+          let alias = Alias.make alias ~dir in
+          let deps = Path.Set.singleton (Path.build stamp_target) in
+          Rules.Produce.Alias.add_deps alias deps
```

Comment:
I guess `add_deps` currently adds the dep both as a "transitive dep" and as a a part of the alias expansion? It seems wrong to add the stamp to alias expansion, but if there's no away to avoid that it's OK to leave it as is (and add a comment, perhaps).

Reply:
I've added a TODO for now. Indeed I don't see any other way.